### PR TITLE
Release 32.0

### PR DESCRIPTION
**Pull request by petersmythe**
# Checklist

- [ ] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [ ] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [ ] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [ ] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [ ] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: docs/developer/procedures/release.rst [hunk 1]
```diff
@@ -136,14 +136,14 @@ When creating the first release candidate of a series, there are some extra step
     git commit -m "Change 28.x to stable branch"
     git push upstream 28.x
 
-  For the new `maintenance` branch, and the remote for the official GeoTools is called ``upstream``::
+  For the new `maintenance` branch (which was the previous `stable` branch)::
   
-    git checkout 26.x
```
Comment (petersmythe) [inline docs/developer/procedures/release.rst:153]: @jodygarnett please check that these changes below are correct.

If 27.x was the stable branch, the new stable branch is 28.x (set above) and then 27.x should become the maintenance branch.  Does the 26.x branch ever get set to archive or similar?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -7,6 +7,13 @@ updates:
     target-branch: 'main'
     open-pull-requests-limit: 10
 
+  - package-ecosystem: 'github-actions'
+    directory: '/'
+    schedule:
+      interval: 'monthly'
+    target-branch: '32.x'
+    open-pull-requests-limit: 10
+
```
Comment (petersmythe) [inline .github/dependabot.yml:16]: Should 30.x below not be deleted i.e. every branch shifts along one version?

File: docs/developer/procedures/release.rst (hunk 1 above)
Comment (jodygarnett) [reply docs/developer/procedures/release.rst:153]: yes, I think that happens further down the page.

File: .github/dependabot.yml (hunk 2 above)
Comment (jodygarnett) [reply .github/dependabot.yml:16]: Yes; although I also wonder why this is not just *main" branch, and then backport dependency change if appropriate? Like our policy is to always commit to "main" 1st and then backport...

Comment (petersmythe) [reply .github/dependabot.yml:16]: That makes sense to me, I will adjust

Comment (mprins) [reply .github/dependabot.yml:16]: No, read the doc plz

Comment (mprins) [reply .github/dependabot.yml:16]: As discussed in the PR and ML each branch gets it's own PRs for these updates

Comment (petersmythe) [reply .github/dependabot.yml:16]: Thanks Mark, are you perhaps talking about this documentation: https://docs.github.com/en/code-security/getting-started/dependabot-quickstart-guide?  It does make sense to have dependabot running on older branches (e.g. maintenance 31.x) as they might have different dependencies being checked, and the dependabot will prepare the appropriate PRs (I am not sure what a ML is).

So then my original question stands: the archive 30.x should be disabled/removed?  I think so.

**Comment by petersmythe** [comment]
Does `backport-32.x` label get created automatically, or does that need to be added to the instructions @jodygarnett ?

File: docs/developer/procedures/release.rst (hunk 1 above)
Comment (petersmythe) [reply docs/developer/procedures/release.rst:153]: Suggested fix: https://github.com/geotools/geotools/pull/4907

File: .github/dependabot.yml (hunk 2 above)
Comment (mprins) [reply .github/dependabot.yml:16]: I apologise for being short previously; dependabot is not configured to check dependencies (apart from security updates by default on the main branch); the idea is to keep the workflows/actions for each branch up-2-date, at some point some of the older branches can be removed from the dependabot.yml config, but we still see commits to branches that have reached EOL when there is a crucial security update

**Pull request merged**